hack/minikube-rook-mirror-setup.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+set -x
+set -e -o pipefail
+
+## Variables
+PRIMARY_CLUSTER="${PRIMARY_CLUSTER:-hub}"
+SECONDARY_CLUSTER="${SECONDARY_CLUSTER:-cluster1}"
+POOL_NAME="replicapool"
+
+## Usage
+usage()
+{
+	set +x
+	echo "Usage:"
+	echo "  $0"
+	echo "  Available environment variables:"
+	echo "    minikube primary cluster PRIMARY_CLUSTER ${PRIMARY_CLUSTER}"
+	echo "    minikube secondary cluster SECONDARY_CLUSTER ${SECONDARY_CLUSTER}"
+	exit 0
+}
+
+function wait_for_condition() {
+    local count=15
+    local condition=${1}
+    local result
+    shift
+
+    while ((count > 0)); do
+        result=$("${@}")
+        if [[ "$result" == "$condition" ]]; then
+            return 0
+        fi
+        count=$((count - 1))
+        sleep 5
+    done
+
+    echo "Failed to meet $condition for command $*"
+    exit 1
+}
+
+SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME=$(kubectl get cephblockpools.ceph.rook.io "${POOL_NAME}" --context="${SECONDARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.info.rbdMirrorBootstrapPeerSecretName}')
+SECONDARY_CLUSTER_SECRET=$(kubectl get secret -n rook-ceph "${SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME}" --context="${SECONDARY_CLUSTER}" -o jsonpath='{.data.token}'| base64 -d)
+SECONDARY_CLUSTER_SITE_NAME=$(kubectl get cephblockpools.ceph.rook.io "${POOL_NAME}" --context="${SECONDARY_CLUSTER}" -nrook-ceph -o jsonpath='{.status.mirroringInfo.site_name}')
+
+echo SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME is "$SECONDARY_CLUSTER_PEER_TOKEN_SECRET_NAME"
+echo Token for the secondary cluster is "$SECONDARY_CLUSTER_SECRET"
+echo SECONDARY_CLUSTER_SITE_NAME is "${SECONDARY_CLUSTER_SITE_NAME}"
+
+kubectl -n rook-ceph create secret generic --context="${PRIMARY_CLUSTER}" "${SECONDARY_CLUSTER_SITE_NAME}" \
+        --from-literal=token="${SECONDARY_CLUSTER_SECRET}" \
+        --from-literal=pool="${POOL_NAME}"
+
+cat <<EOF | kubectl --context="${PRIMARY_CLUSTER}" apply -f -
+apiVersion: ceph.rook.io/v1
+kind: CephRBDMirror
+metadata:
+  name: my-rbd-mirror
+  namespace: rook-ceph
+spec:
+  count: 1
+  peers:
+    secretNames:
+      - "${SECONDARY_CLUSTER_SITE_NAME}"
+EOF
+
+kubectl wait deployments -n rook-ceph --for condition=available rook-ceph-rbd-mirror-a --timeout=60s --context="${PRIMARY_CLUSTER}"
```

Comment:
Creation of secret is imperative style.  Hence, I suggest doing one of the following:
1. Wait for deployment condition prior to creating the secret, so that the script will not fail when rerun.
2. Test for presence of secret prior to attempting to create the secret.
3. Change secret creation to declarative style.

Needless to say that option 2 or 3 is better for making the script idempotent, but option 1 should also be fine for now.